zerver/lib/home.py

```diff
@@ -137,7 +137,7 @@ def build_page_params_for_home_page_load(
         "notification_settings_null": True,
         "bulk_message_deletion": True,
         "user_avatar_url_field_optional": True,
-        "stream_typing_notifications": False,  # Set this to True when frontend support is implemented.
+        "stream_typing_notifications": True,
```

Comment:
Can you start an "api design" conversation posting the current typing notifications API for stream messages just to get eyes on it before we merge this? 

Before we turn this on is a good opportunity to make any changes to improve that API -- E.g., if we hadn't already done it, this would be a good opportunity to rename `private` to `direct` as the message type for these without clients needing to write any compatibility code if we do it before merging the feature.

Reply:
[CZO discussion](https://chat.zulip.org/#narrow/stream/378-api-design/topic/stream.20typing.20indicator/near/1649887)

Reply:
PR: #26958